Build fails on Android

### Description

After upgrading from version 16.1.3 to 17.0.1, my Android build started failing.

```
[17:49:41]: ▸ C/C++: /home/runner/work/app/app/node_modules/react-native-camera-kit/android/build/generated/source/codegen/jni/react/renderer/components/NativeCameraKitSpec/NativeCameraKitSpecJSI-generated.cpp:15:22: error: unknown type name 'NativeCameraKitModuleCxxSpecJSI'
[17:49:41]: ▸ C/C++:    15 |   return static_cast<NativeCameraKitModuleCxxSpecJSI *>(&turboModule)->capture(
[17:49:41]: ▸ C/C++:       |                      ^
[17:49:41]: ▸ C/C++: /home/runner/work/app/app/node_modules/react-native-camera-kit/android/build/generated/source/codegen/jni/react/renderer/components/NativeCameraKitSpec/NativeCameraKitSpecJSI-generated.cpp:15:72: error: no member named 'capture' in 'facebook::react::TurboModule'
[17:49:41]: ▸ C/C++:    15 |   return static_cast<NativeCameraKitModuleCxxSpecJSI *>(&turboModule)->capture(
[17:49:41]: ▸ C/C++:       |                                                         ~~~~~~~~~~~~   ^
[17:49:41]: ▸ C/C++: /home/runner/work/app/app/node_modules/react-native-camera-kit/android/build/generated/source/codegen/jni/react/renderer/components/NativeCameraKitSpec/NativeCameraKitSpecJSI-generated.cpp:22:22: error: unknown type name 'NativeCameraKitModuleCxxSpecJSI'
[17:49:41]: ▸ C/C++:    22 |   return static_cast<NativeCameraKitModuleCxxSpecJSI *>(&turboModule)->requestDeviceCameraAuthorization(
[17:49:41]: ▸ C/C++:       |                      ^
[17:49:41]: ▸ C/C++: /home/runner/work/app/app/node_modules/react-native-camera-kit/android/build/generated/source/codegen/jni/react/renderer/components/NativeCameraKitSpec/NativeCameraKitSpecJSI-generated.cpp:22:72: error: no member named 'requestDeviceCameraAuthorization' in 'facebook::react::TurboModule'
[17:49:41]: ▸ C/C++:    22 |   return static_cast<NativeCameraKitModuleCxxSpecJSI *>(&turboModule)->requestDeviceCameraAuthorization(
[17:49:41]: ▸ C/C++:       |                                                         ~~~~~~~~~~~~   ^
[17:49:41]: ▸ C/C++: /home/runner/work/app/app/node_modules/react-native-camera-kit/android/build/generated/source/codegen/jni/react/renderer/components/NativeCameraKitSpec/NativeCameraKitSpecJSI-generated.cpp:27:22: error: unknown type name 'NativeCameraKitModuleCxxSpecJSI'
[17:49:41]: ▸ C/C++:    27 |   return static_cast<NativeCameraKitModuleCxxSpecJSI *>(&turboModule)->checkDeviceCameraAuthorizationStatus(
[17:49:41]: ▸ C/C++:       |                      ^
[17:49:41]: ▸ C/C++: /home/runner/work/app/app/node_modules/react-native-camera-kit/android/build/generated/source/codegen/jni/react/renderer/components/NativeCameraKitSpec/NativeCameraKitSpecJSI-generated.cpp:27:72: error: no member named 'checkDeviceCameraAuthorizationStatus' in 'facebook::react::TurboModule'
[17:49:41]: ▸ C/C++:    27 |   return static_cast<NativeCameraKitModuleCxxSpecJSI *>(&turboModule)->checkDeviceCameraAuthorizationStatus(
[17:49:41]: ▸ C/C++:       |                                                         ~~~~~~~~~~~~   ^
[17:49:41]: ▸ C/C++: /home/runner/work/app/app/node_modules/react-native-camera-kit/android/build/generated/source/codegen/jni/react/renderer/components/NativeCameraKitSpec/NativeCameraKitSpecJSI-generated.cpp:32:1: error: use of undeclared identifier 'NativeCameraKitModuleCxxSpecJSI'
[17:49:41]: ▸ C/C++:    32 | NativeCameraKitModuleCxxSpecJSI::NativeCameraKitModuleCxxSpecJSI(std::shared_ptr<CallInvoker> jsInvoker)
[17:49:41]: ▸ C/C++:       | ^
[17:49:41]: ▸ C/C++: 7 errors generated.
[17:49:45]: ▸ > Task :app:buildCMakeRelWithDebInfo[arm64-v8a] FAILED
```

React native version: 0.83.0
New architecture enabled

The error occurs in GitHub Actions.

### Steps to reproduce

not sure

### Output of `npx @react-native-community/cli info`

```text
info Fetching system and libraries information...
System:
  OS: macOS 26.3
  CPU: (11) arm64 Apple M3 Pro
  Memory: 1000.86 MB / 18.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 22.14.0
    path: /Users/angelicarosa/.nvm/versions/node/v22.14.0/bin/node
  Yarn:
    version: 1.22.22
    path: /Users/angelicarosa/.nvm/versions/node/v22.14.0/bin/yarn
  npm:
    version: 11.6.4
    path: /Users/angelicarosa/.nvm/versions/node/v22.14.0/bin/npm
  Watchman:
    version: 2024.10.07.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.16.2
    path: /Users/angelicarosa/.rbenv/shims/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 25.2
      - iOS 26.2
      - macOS 26.2
      - tvOS 26.2
      - visionOS 26.2
      - watchOS 26.2
  Android SDK: Not Found
IDEs:
  Android Studio: 2023.1 AI-231.9392.1.2311.11330709
  Xcode:
    version: 26.2/17C52
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 21.0.9
    path: /Library/Java/JavaVirtualMachines/temurin-21.jdk/Contents/Home/bin/javac
  Ruby:
    version: 2.7.4
    path: /Users/angelicarosa/.rbenv/shims/ruby
npmPackages:
  "@react-native-community/cli":
    installed: 20.0.0
    wanted: 20.0.0
  react:
    installed: 19.2.0
    wanted: 19.2.0
  react-native:
    installed: 0.83.0
    wanted: 0.83.0
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: true
iOS:
  hermesEnabled: true
  newArchEnabled: true

info React Native v0.84.0 is now available (your project is running on v0.83.0).
info Changelog: https://github.com/facebook/react-native/releases/tag/v0.84.0
info Diff: https://react-native-community.github.io/upgrade-helper/?from=0.83.0&to=0.84.0
info For more info, check out "https://reactnative.dev/docs/upgrading?os=macos".
```

### MANDATORY Reproducer

missing

### Screenshots and Videos

_No response_

--- a/package.json
+++ b/package.json
@@ -19,6 +19,7 @@
     "release": "yarn clean && yarn build && yarn publish --verbose",
     "release:beta": "yarn clean && yarn build && yarn publish --tag beta --verbose",
     "release:local": "yarn clean && yarn build && tmp=$(mktemp) && yarn pack --filename $tmp.tar.gz && open -R $tmp.tar.gz",
+    "prepack": "rm -rf android/build build",
     "start": "watchman watch-del-all && node node_modules/react-native/local-cli/cli.js start",
     "codegen": "react-native codegen --verbose --path . --platform ios --source library && react-native codegen --verbose --path . --platform android --source library",
     "bootstrap": "cd example/ && bundle install && yarn && cd ios/ && bundle exec pod install",
@@ -31,7 +32,7 @@
   "react-native": "src/index",
   "files": [
     "android",
-    "build",
+    "!android/build",
     "dist",
     "ios",
     "src",

